Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift

```diff
@@ -174,12 +202,9 @@ public extension AlamofireNetworkClient.Request {
     }
   }
   
-  func decode<D: Decodable>(
-    _ decodable: D.Type,
-    decoder: JSONDecoder) -> Promise<D>
-  {
+  func decode<D: Decodable>(_ decodable: D.Type, decoder: JSONDecoder? = nil) -> Promise<D> {
```

Comment:
Why optional decoder?

Reply:
Good question.

I think it makes sense to make it optional which can become handy if you just want to decode the response with the default rules.
On the other hand, making it as required, it's a bit wacky to create a new instance every time when you only need default rules.

What are your concerns?

Reply:
You are creating a new instance inside anyway, right? So why not just
```
func decode<D: Decodable>(
    _ decodable: D.Type,
    decoder: JSONDecoder = .init()) -> Promise<D>
  {
```
?

Reply:
That's another way yes.

Reply:
I prefer this way because the dependency is not _hidden_ inside the method.

Reply:
Right. I've also changed the behavior for the encoder as well.